src/common/core/metacore.js

```diff
@@ -222,12 +222,14 @@ define([
         };
 
         this.getValidPointerNames = function (node) {
-            var validNames = self.getPointerNames(getMetaNode(node)) || [],
+            var metaDefNode = getMetaNode(node),
+                validNames = self.getPointerNames(metaDefNode) || [],
                 i,
                 validPointerNames = [],
                 metaPointerNode, max;
+
             for (i = 0; i < validNames.length; i++) {
-                metaPointerNode = getMetaPointerNode(node, validNames[i]);
+                metaPointerNode = self.getChild(metaDefNode, CONSTANTS.META_POINTER_PREFIX + validNames[i]);
```

Comment:
For the cases I replaced it we already have the `metaDefNode`, plus we are iterating over the `validNames`. The later part means that the check and the additional retrieval of `validNames` in the getMetaPointerNode is redundant.
This is not the case for a places where it is used.